src/webui/hooks/useDeleteReaction.ts

```diff
@@ -0,0 +1,17 @@
+
+
+import johnDoe from '^/domain/authentication/johnDoe';
+import remove from '^/domain/reaction/remove';
+import type ReactionView from '^/domain/reaction/view/ReactionView';
+
+export default function hook(reactions: ReactionView[], setReactions: (reactions: ReactionView[]) => void)
+{
+    return async (reaction: ReactionView) =>
+    {
+        await remove(johnDoe, reaction.id);
+
+        const result = (reactions).filter(item => item.id !== reaction.id);
+
+        setReactions(result);
```

Comment:
```suggestion
        const result = reactions.filter(item => item.id !== reaction.id);

        setReactions(result);

        await remove(johnDoe, reaction.id);
```

I've moved the actual delete below the visual delete to improve interaction and prevent multiple delete attempts when the connection is slow.